streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -529,11 +524,7 @@ void handleLostAll() {
         for (final TaskId id : lockedTaskDirectories) {
             final Task task = tasks.get(id);
             if (task != null) {
-                if (task.isActive() && task.state() == RUNNING) {
-                    taskOffsetSums.put(id, Task.LATEST_OFFSET);
-                } else {
-                    taskOffsetSums.put(id, sumOfChangelogOffsets(id, task.changelogOffsets()));
-                }
+                taskOffsetSums.put(id, sumOfChangelogOffsets(id, task.changelogOffsets()));
```

Comment:
Make TM agnostic to task state -- putting some more logic into `sumOfChangelogOffsets` to make this work